include/meta/meta.hpp

```diff
@@ -2125,6 +2125,21 @@ namespace meta
             using min = defer<min, T, U>;
         }
 
+        /// Index of the first occurrence of the type \p T within the list \p List.
+        /// \par Complexity
+        /// \f$ O(N) \f$.
+        /// \ingroup list
+        template <typename T, typename List>
```

Comment:
No, but I *did* kind of wish for monads when I was implementing `cartesian_product`.